pkg/devfile/adapters/kubernetes/utils/utils.go

```diff
@@ -252,3 +253,39 @@ func overrideContainerArgs(container *corev1.Container) {
 	container.Command = append(container.Command, adaptersCommon.SupervisordBinaryPath)
 	container.Args = append(container.Args, "-c", adaptersCommon.SupervisordConfFile)
 }
+
+// UpdateContainerWithEnvFrom populates the runtime container with relevant
+// values for "EnvFrom" so that component can be linked with Operator backed
+// service
+func UpdateContainerWithEnvFrom(containers []corev1.Container, devfile devfileParser.DevfileObj, devfileRunCmd string, ei envinfo.EnvSpecificInfo) ([]corev1.Container, error) {
+	runCommand, err := adaptersCommon.GetRunCommand(devfile.Data, devfileRunCmd)
```

Comment:
can we send the runCommand's `Component` instead of the devfileObj, sending such a big object for a name comparison will make it hard to test

Reply:
I don't know. I have literally re-used the bits from https://github.com/openshift/odo/blob/fdea76febcf39cc8bca30588989ee3a24b4265af/pkg/devfile/adapters/kubernetes/utils/utils.go#L128-L131 since we care about run container here. 

Where can I know more about this `runCommand` that you're talking about? I found this https://github.com/openshift/odo/blob/fdea76febcf39cc8bca30588989ee3a24b4265af/pkg/devfile/parser/data/common/types.go#L23 but I don't think it's relevant.

Reply:
` adaptersCommon.GetRunCommand(devfile.Data, devfileRunCmd)`, gets you the run command from devfile data, Ideally we should pass command list and it should return `runCommand` from it, but there is lots of complicated logic which needs whole devfile data (components and command) to get `runCommand`. 
IMO It could be refactored later on, right now using it as it is ok.